test/L2/L2AirdropV2.t.sol

```diff
@@ -0,0 +1,594 @@
+    }
+
+    function test_SatisfiesStakingTier2_ZeroRecipientAddress() public {
+        vm.expectRevert("L2AirdropV2: recipient is the zero address");
+        l2AirdropV2.satisfiesStakingTier2(address(0x0), 0);
+    }
+
+    function test_SatisfiesStakingTier2_ZeroAmount() public {
+        vm.expectRevert("L2AirdropV2: airdrop amount is zero");
+        l2AirdropV2.satisfiesStakingTier2(alice, 0);
+    }
+
+    function aliceClaimAirdropForStakingTier1() internal {
+        // alice satisfies staking tier 1 condition
+        aliceSatifiesStakingTier1();
+
+        // alice did not claim airdrop for staking tier 1 condition
+        assertEq(l2AirdropV2.claimedStakingTier1(aliceLSKAddress), false);
+
+        // claim airdrop for alice (only staking tier 1 condition is satisfied)
+        uint256 aliceBalanceBefore = l2LiskToken.balanceOf(alice);
+        bytes32[] memory merkleProof = new bytes32[](1);
+        merkleProof[0] = bytes32(0xf0df3dcda05b4fbd9c655cde3d5ceb211e019e72ec816e127a59e7195f2cd7f5);
+        l2AirdropV2.claimAirdrop(aliceLSKAddress, 80 * 10 ** 18, merkleProof);
+        assertEq(l2LiskToken.balanceOf(alice), aliceBalanceBefore + 40 * 10 ** 18); // 4 L2LiskToken airdrop
+
+        // check that alice has claimed airdrop for staking tier 1 condition
+        assertEq(l2AirdropV2.claimedStakingTier1(aliceLSKAddress), true);
+    }
+
+    function test_ClaimAirdrop_StakingTier1() public {
+        // check that alice can claim airdrop for staking tier 1 condition
+        aliceClaimAirdropForStakingTier1();
+    }
+
+    function aliceClaimAirdropForStakingTier2() internal {
+        // alice satisfies staking tier 2 condition
+        aliceSatifiesStakingTier2();
+
+        // alice did not claim airdrop for staking tier 2 condition
+        assertEq(l2AirdropV2.claimedStakingTier2(aliceLSKAddress), false);
+
+        // claim airdrop for alice (only staking tier 2 condition is satisfied)
+        uint256 aliceBalanceBefore = l2LiskToken.balanceOf(alice);
+        bytes32[] memory merkleProof = new bytes32[](1);
+        merkleProof[0] = bytes32(0xf0df3dcda05b4fbd9c655cde3d5ceb211e019e72ec816e127a59e7195f2cd7f5);
+        l2AirdropV2.claimAirdrop(aliceLSKAddress, 80 * 10 ** 18, merkleProof);
+        assertEq(l2LiskToken.balanceOf(alice), aliceBalanceBefore + 40 * 10 ** 18); // 4 L2LiskToken airdrop
+
+        // check that alice has claimed airdrop for staking tier 2 condition
+        assertEq(l2AirdropV2.claimedStakingTier2(aliceLSKAddress), true);
+    }
+
+    function test_ClaimAirdrop_StakingTier2() public {
+        // first alice will claim airdrop for staking tier 1 condition that only staking tier 2 condition will be left
+        aliceClaimAirdropForStakingTier1();
+
+        // check that alice can claim airdrop for staking tier 2 condition
+        aliceClaimAirdropForStakingTier2();
+    }
+
+    function test_ClaimAirdrop_StakingTier1_StakingTier2() public {
```

Comment:
this test could also be removed as everything is repeated in test_ClaimAirdrop_FullAirdrop